### DISCUSSION TITLE
[SOLVED] Import all youtube videos from a channel

### DISCUSSION THREAD

**Discussion by formlessdao** [martinrotter/rssguard#526] [Q&A]
Is it possible to import all youtube videos from a channel into Rssguard?

**Comment by martinrotter** [comment]
Youtube offers RSS feeds for channels out of box.

https://webapps.stackexchange.com/questions/111680/how-to-find-channel-rss-feed-on-youtube

**Reply by formlessdao** [reply]
I am interested in importing ALL VIDEOS that exist on every channel and sort them by date so i have full history by date of everything on those channels.
I also want to be able to search in all history of those channels from RSSguard by title and also by what's mentioned in the youtube description for every video.

**Reply by martinrotter** [reply]
I see. Well, if that is the case, you probably must use Youtube API, specifically "Search/list" method which returns list of all video IDs of a channel. Then you need to post-process the resulting JSON and scrape it into RSS XML which can then be read by RSS Guard.

All of it can be made automatic via Python script with RSS Guard's feature "website scraping".

**Reply by martinrotter** [reply]
Here is a Python script, you only need to enter API key which you can generate in Google.

```python
# -*- coding: utf-8 -*-

# Sample Python code for youtube.search.list
# See instructions for running these code samples locally:
# https://developers.google.com/explorer-help/guides/code_samples#python

import os

import google_auth_oauthlib.flow
import googleapiclient.discovery
import googleapiclient.errors

scopes = ["https://www.googleapis.com/auth/youtube.force-ssl"]


def main():
    # Disable OAuthlib's HTTPS verification when running locally.
    # *DO NOT* leave this option enabled in production.
    os.environ["OAUTHLIB_INSECURE_TRANSPORT"] = "1"

    api_service_name = "youtube"
    api_version = "v3"

    # Get credentials and create an API client
    youtube = googleapiclient.discovery.build(
        api_service_name,
        api_version,
        developerKey='<KEY>')

    request = youtube.search().list(
        part="snippet",
        channelId="<CHANNEL-ID>",
        order="date"
    )
    response = request.execute()

    print(response)


if __name__ == "__main__":
    main()

```

**Reply by martinrotter** [reply]
response can be further parse in py file and processed into rss or json feed

**Reply by formlessdao** [reply]
Thank you. Where do I need to input API key? also where where do i search for it? I'm not a developer.

**Reply by martinrotter** [reply]
` developerKey='<KEY>')`

You need to generate your key at - https://console.cloud.google.com/apis/credentials

perhaps google some tutorial. Also, once you actually get this step running, there are some additional things you need to do to generate valid RSS/JSON feed, but I guess if you take it one step a time, you can achieve what you want quickly.
